### PR TITLE
Make golint happier

### PR DESCRIPTION
**Pull request by corny**
There is still some more work to do

### CODE REVIEW

File: pkg/api/alerting.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func GetAlertStatesForDashboard(c *m.ReqContext) Response {
 	dashboardID := c.QueryInt64("dashboardId")
 
 	if dashboardID == 0 {
-		return ApiError(400, "Missing query parameter dashboardId", nil)
+		return Error(400, "Missing query parameter dashboardId", nil)
```
Comment (bergquist) [inline pkg/api/alerting.go]: Why should `Api` be removed?

Comment (corny) [reply pkg/api/alerting.go]: `golint pkg/api/... | grep stutt` gives:
```
pkg/api/common.go:79:6: func name will be used as api.ApiSuccess by other packages, and that stutters; consider calling this Success
pkg/api/common.go:85:6: func name will be used as api.ApiError by other packages, and that stutters; consider calling this Error
```

Comment (bergquist) [reply pkg/api/alerting.go]: Makes sense :)

Comment (corny) [reply pkg/api/alerting.go]: It's also more consistent because other similar methods don't have the `Api` prefix.

**Comment by bergquist** [comment]
Thank you!

Right now we don't enforce golint so pull requests like this is an endless battle right now :) 
We might enforce golint in the future. You can track that discussion on https://github.com/grafana/grafana/issues/10381

File: pkg/api/alerting.go (hunk 1 above)
Comment (torkelo) [reply pkg/api/alerting.go]: hm.. then rename it to JsonError or NewErrorRsp, the api prefix is to make it clear the response is json and is only to be used in `/api/` http handlers.  the api package is not only for the http api but also some other http handlers (maybe the package should be named http but that conflicts with standard goland package name)

Comment (corny) [reply pkg/api/alerting.go]: Well, to be consistent, we had to rename several methods to `New***Response` that returns a new `*NormalResponse` instance. Thanks for merging.

**Pull request merged**